### ISSUE TITLE
[SDK52] Unable to build development build for android with new architecture enabled. KSP version issue

### ISSUE THREAD

**Issue by rranjan14** [expo/expo#32535]
### Minimal reproducible example

N/A

### What platform(s) does this occur on?

Android

### Where did you reproduce the issue?

in a development build

### Summary

<img width="904" alt="Screenshot 2024-11-01 at 21 25 10" src="https://github.com/user-attachments/assets/60a0364f-23f8-41c2-a05e-4091d222a805">
<img width="1152" alt="Screenshot 2024-11-01 at 21 25 21" src="https://github.com/user-attachments/assets/1c4d45ac-aee1-4bf9-b126-47a3f72677cf">

ksp version is too old for the kotlin version being used. Happening with `expo-updates`.

### Environment

```text
System:
      OS: macOS 15.1
      Shell: 3.7.1 - /opt/homebrew/bin/fish
    Binaries:
      Node: 20.13.1 - ~/.nvm/versions/node/v20.13.1/bin/node
      Yarn: 4.5.0 - ~/.nvm/versions/node/v20.13.1/bin/yarn
      npm: 10.8.0 - /opt/homebrew/bin/npm
      Watchman: 2024.10.21.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.14.3 - /Users/rahulranjan/.rvm/gems/ruby-2.7.8/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 24.1, iOS 18.1, macOS 15.1, tvOS 18.1, visionOS 2.1, watchOS 11.1
      Android SDK:
        API Levels: 33, 34, 35
        Build Tools: 30.0.2, 30.0.3, 33.0.0, 34.0.0, 35.0.0, 35.0.0
        System Images: android-34 | Google APIs ARM 64 v8a, android-VanillaIceCream | Google Play ARM 64 v8a
    IDEs:
      Android Studio: 2024.2 AI-242.23339.11.2421.12550806
      Xcode: 16.1/16B40 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^52.0.0-preview.18 => 52.0.0-preview.18
      react: 18.3.1 => 18.2.0
      react-native: 0.76.1 => 0.76.1
    npmGlobalPackages:
      eas-cli: 12.6.0
    Expo Workflow: bare
```

### Expo Doctor Diagnostics

```text
Enabled experimental React Native Directory checks. Unset the EXPO_DOCTOR_ENABLE_DIRECTORY_CHECK environment variable to disable this check.
✔ Check package.json for common issues
✖ Check Expo config for common issues
✔ Check dependencies for packages that should not be installed directly
✔ Check for issues with Metro config
✔ Check npm/ yarn versions
✔ Check native tooling versions
✔ Check if the project meets version requirements for submission to app stores
✔ Check for app config fields that may not be synced in a non-CNG project
✔ Check for common project setup issues
✖ Validate packages against React Native Directory package metadata
✖ Check Expo config (app.json/ app.config.js) schema
✖ Check that packages match versions required by installed Expo SDK
✔ Check for legacy global CLI installed locally
✔ Check that native modules do not use incompatible support packages
✔ Check that native modules use compatible support package versions for installed Expo SDK

Detailed check results:

Error: Problem validating fields in /Users/rahulranjan/temp/wokay/apps/mobile/app.json. Learn more: https://docs.expo.dev/workflow/configuration/
 • should NOT have additional property 'hooks'.

You have an app.json file in your project, but your app.config.ts is not using the values from it. Remove the static app.json, or use its values in your dynamic app.config.js. Learn more: https://docs.expo.dev/workflow/configuration

The following issues were found when validating your dependencies against React Native Directory:
Unsupported on New Architecture: react-native-image-crop-picker
Untested on New Architecture: lucide-react-native
Unmaintained: @gorhom/portal, react-native-section-list-get-item-layout
No metadata available: @config-plugins/react-native-blob-util, @config-plugins/react-native-pdf, @dev-plugins/react-native-mmkv, @dev-plugins/react-navigation, @webview-bridge/react-native, @xstate/react, eventemitter3, flexsearch, jotai, lodash.throttle, mime, pretty-bytes, react-native-gradient-shimmer, react-native-ios-utilities
Advice:
- Use libraries that are actively maintained and support the New Architecture. Find alternative libraries with https://reactnative.directory.
- Add packages to expo.doctor.reactNativeDirectoryCheck.exclude in package.json to selectively skip validations, if the warning is not relevant.
- Update React Native Directory to include metadata for unknown packages. Alternatively, set expo.doctor.reactNativeDirectoryCheck.listUnknownPackages in package.json to false to skip warnings about packages with no metadata, if the warning is not relevant.

The following packages should be updated for best compatibility with the installed expo version:
  react@18.2.0 - expected version: 18.3.1
Your project may not work correctly until you install the expected versions of the packages.
Found outdated dependencies
Advice: Use 'npx expo install --check' to review and upgrade your dependencies.

One or more checks failed, indicating possible issues with the project.
```

**Comment by kontea-dev** [comment]
Hello , i've the same issue . Have you found any solution ?

**Comment by Giulio987** [comment]
Hi, I've encountered the same issue:

```bash
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':expo-updates:bundleLibCompileToJarDebug'.
> Could not create task ':expo-updates:kspDebugKotlin'.
   > Cannot query the value of task ':expo-updates:compileDebugKotlin' property 'moduleName' because it has no value available.
```

My package versions:
```json
"expo": "~52.0.26",
"expo-updates": "~0.26.13"
```
Configuration in expo-build-properties
To ensure the correct configuration for expo-build-properties, I used the following setup:

```js
[
  'expo-build-properties',
  {
    android: {
      compileSdkVersion: 35,
      targetSdkVersion: 35,
      buildToolsVersion: '35.0.0',
      kotlinVersion: '1.9.0',
    },
    ios: {
      deploymentTarget: '15.1',
    },
  },
];
```

**Fix**: _Adding the kspVersion_
The issue seems to be related to the kspVersion. I added this to the Android configuration to resolve the problem:

```js
android: {
  compileSdkVersion: 35,
  targetSdkVersion: 35,
  buildToolsVersion: '35.0.0',
  kotlinVersion: '1.9.0',
  kspVersion: '1.9.20-1.0.14', // <-- ADD THIS
},
```
I hope this helps anyone facing the same problem!

**Comment by yuchida0911** [comment]
```
"expo": "52.0.19",
"expo-updates": "~0.27.1"
```
I was also facing a similar issue. Although I was able to build an app, it crashes right on launch. 

in my case, all I had to do was to set kotlinVersion to "1.9.24", which seems like to set the `kspVersion` in `expo-modules-core` (build.gradle) to `1.9.24-1.0.20`

![Image](https://github.com/user-attachments/assets/15f93b3e-3600-475b-b678-eacadf3423d1)

Previously, my project did not specify `kotlinVersion` hence it was set to 1.6.10 by default, which end up using old kspVersion.


``` 
android: {
  ...
  "kotlinVersion": '1.9.24',
},
```
I hope it might help someone.
